main.go

```diff
@@ -125,11 +147,38 @@ func main() {
 		pingers[i] = pinger
 	}
 
+	sc.Lock()
+	for _, targetGroup := range sc.C.Targets {
+		for _, host = range targetGroup.Hosts {
+			pinger = ping.New(host)
+			pinger.Interval = targetGroup.Interval
+			pinger.SetNetwork(targetGroup.Network)
+			if targetGroup.Protocol == "icmp" {
+				pinger.SetPrivileged(true)
+			}
+			err := pinger.Resolve()
+			if err != nil {
+				log.Errorf("failed to resolve pinger: %s\n", err.Error())
+				return
+			}
+			pingers = append(pingers, pinger)
+		}
+	}
+	sc.Unlock()
+
+	if len(pingers) == 0 {
+		log.Errorf("no targets specified on command line or in config file")
+		return
+	}
+
 	splay := time.Duration(interval.Nanoseconds() / int64(len(pingers)))
 	log.Infof("Waiting %s between starting pingers", splay)
+	g := new(errgroup.Group)
 	for _, pinger := range pingers {
 		log.Infof("Starting prober for %s", pinger.Addr())
-		go pinger.Run()
+		g.Go(func() error {
```

Comment:
Is this safe? Can pinger be different when we call it ? Maybe g.Go(pinger.Run) would be safer?

Reply:
Oh yea, I forget why I did it that way. This is better.